Incorporate feedback from the Ethical ML workshops

anssiko opened the pull request:
- Update Operationalization: Putting the Principles into Practice
- Add Register of Risks and Mitigations


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/ethical-webmachinelearning/pull/20.html" title="Last updated on Jun 2, 2022, 1:31 PM UTC (a272f56)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/ethical-webmachinelearning/20/25a8ba5...a272f56.html" title="Last updated on Jun 2, 2022, 1:31 PM UTC (a272f56)">Diff</a>

anssiko committed:
Incorporate feedback from the Ethical ML workshops

- Update Operationalization: Putting the Principles into Practice
- Add Register of Risks and Mitigations

anssiko commented:
Gentle ping, PTAL @dontcallmedom :-)

Some areas to focus on, mostly presentational:

- Proposals welcome on how to better represent Risks -> Possible Mitigations mapping. Maybe with some CSS tricks we could use a two-column layout for Risks and Possible Mitigations side-by-side (when screen estate allows)? Someone with proper CSS-fu would need to help with that. Or we could consider that future work.

- "This bullet intentionally left blank" in Possible Mitigations. I was thinking of turning these empty bullets into links that open a new GH issue to encourage contributions.

Comments:

- I'd prefer to keep the source format as simple as possible and let CSS handle presentational aspects, if possible. Complex tables are tedious to edit.

- To help editing the lists and correlate Risks with Possible Mitigations in source, maybe instead of:
```
1.

1.

1.
```

We should use:

```
1.

2.

3.
```

dontcallmedom commented on index.bs:
```diff
@@ -442,40 +442,474 @@ Where tensions arise between the principles above, they should be resolved throu
 
+So for example, if we consider the principle of “Fairness and non-dicrimination”, a risk might be that biases in training data lead to model predictions that are less accurate for particular groups, resulting in real-world harms (e.g. denial of services). A mitigation might be to test the training data and model properly for bias, or indeed it might be to not make access to essential services dep ...
+
+To provide a more structured approach to this, we have developed a workshop format which you can adapt and use.
```
```suggestion
To provide a more structured approach to this, we have developed a workshop format that can be adapted and used to help identify such risks & mitigations.
```

dontcallmedom replied:
I wonder if the details about the workshop (until "it's worth noting" below) should be moved to an appendix - while very useful, it feels a bit out of place in the flow of the document.

anssiko replied:
Moved to appendix. Text changes integrated into https://github.com/webmachinelearning/ethical-webmachinelearning/pull/20/commits/13d34dc490bdea3ce0341c4b39f26eb2000c206e

dontcallmedom commented on index.bs:
```diff
@@ -442,40 +442,474 @@ Where tensions arise between the principles above, they should be resolved throu
+
+To provide a more structured approach to this, we have developed a workshop format which you can adapt and use.
+
+The workshop is in two parts (please make a copies for your own use):
```
```suggestion
The workshop is articulated around two documents that are expected to be completed interactively by participants:
```

anssiko replied:
Integrated into https://github.com/webmachinelearning/ethical-webmachinelearning/pull/20/commits/13d34dc490bdea3ce0341c4b39f26eb2000c206e

dontcallmedom commented on index.bs:
```diff
@@ -442,40 +442,474 @@ Where tensions arise between the principles above, they should be resolved throu
+Part One - [Ethical thinking workshop](https://docs.google.com/document/d/1f_PcByjW8-zXbYWeEyOl-RpZ3SKapm_MWSaNZ9ZOi4c/edit?usp=sharing) - is about using the principles to generate and prioritize potential risks
+
+Part Two - [Ethical Risk Canvas](https://docs.google.com/document/d/1hTQnpWC5KC4qIJB9-Kkd46yMVgDuTCQlA3MffqtUbCI/edit?usp=sharing) - is about digging deeper into specific risks and thinking about who they might impact and how best to mitigate them.
+
```
we should probably post a PDF copy of these documents somewhere if only for archival sake.

anssiko replied:
Opened https://github.com/webmachinelearning/ethical-webmachinelearning/issues/21

dontcallmedom commented on index.bs:
```diff
@@ -442,40 +442,474 @@ Where tensions arise between the principles above, they should be resolved throu
+
+# Register of Risks and Mitigations
+
+Note: The risk register is work in progress and welcomes further review and tidying up.
```
I'm happy to leave this for a future pull request, but I think there is indeed substantive work to harmonize the phrasing of the risks, with possibly merging / sorting some of them as well.

dontcallmedom commented on index.bs:
```diff
@@ -442,40 +442,474 @@ Where tensions arise between the principles above, they should be resolved throu
+
+### Possible Mitigations
+
+1. Browser-assisted mechanisms to find out about the limitations and performance characteristics of ML models used in a Web app. This could build on an approach published in Model Cards for Model Reporting where making this report machine-discoverable would allow for the web browser to offer a more integrated user experience. Another transparency tool is the [Open Ethics Transparency Protocol](ht ...
```
In terms of mapping mitigations to risks, I would use a naming scheme for risks (e.g. FND-R1 for risk 1 under **F**airness and **N**on-**D**iscrimination), and then references these names from the mitigations that address them.

anssiko replied:
Done in https://github.com/webmachinelearning/ethical-webmachinelearning/pull/20/commits/b00171e56c8d40f936040d19c877b5fccc972aba

anssiko committed:
Move Workshop Format and Templates to Appendix

Incorporate review feedback from Dom

anssiko committed:
Implement a naming scheme for risks to mitigations mapping

- Creates dfns for all risks and mitigations
- Also adds Bikeshed bikeshed includes for repeated
  notes/issue blocks

anssiko committed:
Add Bikeshed includes

anssiko committed:
Update Bikeshed issue include

anssiko commented:
Gentle ping, PTAL @dontcallmedom.

Your feedback has been addressed. I'd like to merge this after your approval so the WG can advance with its W3C Note publication plan.

anssiko commented:
TY @dontcallmedom!
